napalm/nxos/nxos.py

```diff
@@ -632,9 +632,13 @@ def _parse_vlan_ports(self, vlan_s):
             find = re.findall(find_regexp, vls.strip())
             if find:
                 for i in range(int(find[0][1]), int(find[0][2]) + 1):
-                    vlans.append(find[0][0] + str(i))
+                    vlans.append(
+                        napalm.base.helpers.canonical_interface_name(
+                            find[0][0] + str(i)
+                        )
+                    )
             else:
-                vlans.append(vls.strip())
+                vlans.append(napalm.base.helpers.canonical_interface_name(vls.strip()))
```

Comment:
I like this, just wanted to mention that on IOS at least there's an optional argument `canonical_int` defaulting to `False` currently - is this something we'd want on NX-OS as well? My preference would probably be to always use the canonical map, but maybe there are some reasons on the existence of `canonical_int` arg I'm not aware of. CC @ktbyers

Reply:
I do not see this anywhere else in the nxos/nxos_ssh code. I can add it if desired.

Reply:
Let's open that as a separate issue (so this PR doesn't get bogged down). I would guess for backwards compatibility, but it doesn't look like it was fully/consistently implemented.

I would vote always use canonical interface and just get rid of the argument to toggle that off (especially given that it looks like it is currently inconsistently implemented).